### PR TITLE
[processor/resourcedetection] Add GKE detector

### PR DESCRIPTION
**Pull request by jrcamp**
This adds an gke detector for GKE. It sets cloud.provider,
cloud.infrastructure_service, and k8s.cluster.name.

It's intended to be used in conjunction with gce detector. The gke detector
should come first so that the more specific cloud.infrastructure_service value
wins (see README.md).

Note that this code is taken from
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/detectors/gcp.
There is a goal to prevent duplication with go-contrib in #1590 and I started
down that path but it will require more time than currently available as it
needs to transform from different data models, deal with any differences in
metadata between existing detectors, etc.

### CODE REVIEW

File: processor/resourcedetectionprocessor/internal/gcp/gke/gke.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	// Check if on GCP.
+	if !gke.metadata.OnGCE() {
+		return res, nil
+	}
+
+	attr := res.Attributes()
+	attr.InsertString(conventions.AttributeCloudProvider, conventions.AttributeCloudProviderGCP)
+
+	// Check if running on k8s.
+	if os.Getenv("KUBERNETES_SERVICE_HOST") == "" {
```
Comment (tigrannajaryan) [inline processor/resourcedetectionprocessor/internal/gcp/gke/gke.go:57]: What's this env var? Please define as const / document.

File: processor/resourcedetectionprocessor/internal/gcp/gke/gke.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		return res, nil
+	}
+
+	attr.InsertString(conventions.AttributeCloudInfrastructureService, conventions.AttributeCloudProviderGCPGKE)
+
+	if clusterName, err := gke.metadata.InstanceAttributeValue("cluster-name"); err != nil {
```
Comment (tigrannajaryan) [inline processor/resourcedetectionprocessor/internal/gcp/gke/gke.go:63]: Same for "cluster-name": define as const / document.

File: processor/resourcedetectionprocessor/internal/gcp/metadata.go [hunk 3]
```diff
@@ -4,54 +4,61 @@
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
+//       http://www.apache.org/licenses/LICENSE-2.0
```
Comment (tigrannajaryan) [inline processor/resourcedetectionprocessor/internal/gcp/metadata.go:7]: Is this change intentional?

**Review by tigrannajaryan — APPROVED** [review]

File: processor/resourcedetectionprocessor/internal/gcp/metadata.go (hunk 3 above)
Comment (jrcamp) [reply processor/resourcedetectionprocessor/internal/gcp/metadata.go]: Not sure why or how it got changed.. reverted.

**Review by aabmass — APPROVED** [review]
I was able to verify this working in a GKE cluster, so LGTM. @punya or @dashpole any other comments?

```
Resource labels:
     -> cloud.provider: STRING(gcp)
     -> cloud.infrastructure_service: STRING(gcp_gke)
     -> k8s.cluster.name: STRING(my-first-cluster-1)
     -> cloud.account.id: STRING(otel-starter-project)
     -> cloud.zone: STRING(us-central1-c)
     -> host.name: STRING(gke-my-first-cluster-1-default-pool-567cf672-3bsv.c.otel-starter-project.internal)
     -> host.id: STRING(7389811979834592377)
     -> host.type: STRING(projects/370301448343/machineTypes/g1-small)
```

**Comment by bogdandrutu** [comment]
Please fix tests

**Comment by jrcamp** [comment]
@bogdandrutu fixed

**Review by bogdandrutu — APPROVED** [review]
